src/track/track.cpp

```diff
@@ -952,13 +969,23 @@ bool Track::importPendingBeatsWhileLocked() {
 }
 
 void Track::importPendingBeatsMarkDirtyAndUnlock(
-        QMutexLocker* pLock) {
+        QMutexLocker* pLock,
+        bool bpmLocked) {
     DEBUG_ASSERT(pLock);
+
+    if (m_record.getBpmLocked()) {
```

Comment:
Dangerous. This should never happen. Silently doing nothing is not a good idea.

Reply:
Now we have a return value and the function has a try prefix.